apps/rewards/src/index.ts

```diff
@@ -51,108 +73,118 @@ const claimRewards = async (env: Env, provider: any) => {
   );
 };
 
+interface ZeroXData {
+  buyAmount: string;
+  data: string;
+}
 const getZeroExTradeData = async (
   env: Env,
   sellToken: string,
   buyToken: string,
   amount: BigNumber
-) => {
+): Promise<ZeroXData> => {
   const params = `?sellToken=${sellToken}&buyToken=${buyToken}&sellAmount=${amount.toString()}&slippagePercentage=0.05`;
+  console.log("params", params);
   const resp = await fetch(env.ZERO_EX_PRICE_URL + params, {
     headers: {
       '0x-api-key': env.ZERO_EX_API_KEY,
     },
   });
 
   // Delay to prevent rate limiting
-  await new Promise((f) => setTimeout(f, 2000));
+  await wait(2000);
 
   return resp.json();
 };
 
+type FunRetProm = () => Promise<any>;
+const executePromisesSequentially = async (funcArray: FunRetProm[]): Promise<any[]> => {
+  return funcArray.reduce(async (accumulator, func) => {
+    const results = await accumulator;
+    return [...results, await func()];
+  }, Promise.resolve([]));
+};
+
+const getTrades = async (
+  env: Env,
+  sellToken: string,
+  tokens: string[],
+  sellAmounts: BigNumber[]
+) => {
+  // we need to execute them sequentially due to rate limit on 0x api
+  return executePromisesSequentially(tokens.map((token, i) => async () => {
+    const amount = sellAmounts[i];
+    let oracleSlippagePercentOrLimit = '0';
+    let exchangeData = '0x00';
+
+    if (!amount.eq(0)) {
+      let retryCount = 0;
+      let tradeData: any;
+      do {
+        tradeData = await getZeroExTradeData(env,
+          sellToken,
+          token,
+          amount
+        );
+        if (tradeData.buyAmount == undefined) {
+          console.log('Request to 0x failed: ', retryCount);
+        }
+        // 0x API sometimes returns  insufficient  liquidity error
+        // so we retry here
+      } while (tradeData.buyAmount == undefined && retryCount++ < 5);
+
+      oracleSlippagePercentOrLimit =
+        BigNumber.from(tradeData.buyAmount).mul(95).div(100).toString();
+      exchangeData = tradeData.data;
+    }
+
+    return [
+      sellToken,
+      token == ARB_WETH ? "0x0000000000000000000000000000000000000000" : token,
```

Comment:
It might be cleaner to specify '0x00000..." as ARB_ETH and then whitelist it into the rETH/WETH pool since Balancer handles ETH / WETH conversion automatically. In the future since some Curve pools do not do this for you we may need to ask the contract to sell for WETH explicitly.